File: JULY-2025/09-07-2025/p3439.scala
object Solution {
    def maxFreeTime(eventTime: Int, k: Int,
        startTime: Array[Int], endTime: Array[Int]): Int = {
        val array = (startTime :+ eventTime).zip(0 +: endTime).map(v => v._1 - v._2)
        val sum = array.take(k + 1).sum
        array.drop(k + 1).foldLeft((sum, sum, 0)) {
            case ((max, currentSum, i), value) =>
                val newSum = currentSum - array(i) + value
                (math.max(max, newSum), newSum, i + 1)
        }._1
    }
}

File: JULY-2025/09-07-2025/p3439.rs
impl Solution {

    pub fn max_free_time(evt: i32, k: i32, st: Vec<i32>, et: Vec<i32>) -> i32 {
        let (mut sum, k) = (0, k as usize);
        (0..=st.len()).map(|i| {
            sum += if i < st.len() { st[i] } else { evt } - if i > 0 { et[i - 1] } else { 0 } -
            if i > k { st[i - k - 1] } else { 0 } + if i >= k + 2 { et[i - k - 2] } else { 0 };
            sum
        }).max().unwrap_or(0)
    }

}